cdap-security/src/main/java/io/cdap/cdap/security/auth/UserIdentity.java

```diff
@@ -30,14 +30,20 @@
  */
 public class UserIdentity {
   static final class Schemas {
-    private static final int VERSION = 1;
+    private static final int VERSION = 2;
     private static final Map<Integer, Schema> schemas = Maps.newHashMap();
     static {
       schemas.put(1, Schema.recordOf("AccessTokenIdentifier",
```

Comment:
What would be `identifierType` for schema version 1? If it's null, you should handle it

Reply:
Not too clear on how to handle a null enum when the class is instantiated by reflection; I put null checks on the calls to getIdentifierType, but I can also put a null check inside of the getIdentifierType method and add an INVALID type for null types.